README.md

```diff
@@ -1 +1,134 @@
+
+There are two python scripts in this repository:
+- [G2Snapshot.py](G2Snapshot.py) is the latest version that interacts directly with the database and is multi-threaded 
+for improved performance on large data sets.
+- [G2Snapshot-api-only.py](G2Snapshot-api-only.py) is the legacy version that only uses the API.  It does not require direct database access, 
+but may not perform well on larger data sets.
+
+*You will want to install database access as described in the prerequisites below and use G2Snapshot.py on large databases.  If you do not
+have database access, you will have to use the G2Snapshot-api-only version which runs slower and has less functionality.*
+
+Usage:
+
+```console
+python3 G2Snapshot.py --help
+usage: G2Snapshot.py [-h] [-c CONFIG_FILE_NAME] [-o OUTPUT_FILE_ROOT]
+                     [-s SAMPLE_SIZE] [-d DATASOURCE_FILTER]
+                     [-f RELATIONSHIP_FILTER] [-a] [-D] [-k CHUNK_SIZE]
+                     [-t THREAD_COUNT] [-u]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -c CONFIG_FILE_NAME, --config_file_name CONFIG_FILE_NAME
+                        name of the senzing config file, defaults to
+                        /etc/opt/senzing/G2Module.ini
+  -o OUTPUT_FILE_ROOT, --output_file_root OUTPUT_FILE_ROOT
+                        root name for files created such as
+                        "/project/snapshots/snapshot1"
+  -s SAMPLE_SIZE, --sample_size SAMPLE_SIZE
+                        defaults to 1000
+  -f RELATIONSHIP_FILTER, --relationship_filter RELATIONSHIP_FILTER
+                        filter options 1=No Relationships, 2=Include possible
+                        matches, 3=Include possibly related and disclosed.
+                        Defaults to 3
+  -a, --for_audit       export csv file for audit
+  -D, --debug           print debug statements
+```
+*the following are not in G2Snapshot-api-only*
+```console
+  -d DATASOURCE_FILTER, --datasource_filter DATASOURCE_FILTER
+                        data source code to analayze, defaults to all
+  -k CHUNK_SIZE, --chunk_size CHUNK_SIZE
+                        defaults to 1000000
+  -t THREAD_COUNT, --thread_count THREAD_COUNT
+                        defaults to 0
+  -u, --use_api         use api instead of sql to get resume
+```
+
+## Contents
+
+1. [Prerequisites](#Prerequisites)
+2. [Installation](#Installation)
+3. [Typical use](#Typical-use)
+
+### Prerequisites
+- Python 3.6 or higher
+- Senzing API version 1.10 or higher
+- database support for your Senzing database
+    - psycopg2 (for postgres)
+    - pyodbc (and supporting drivers) for all other databases 
+
+*If using an SSHD container, you should first max it out with at least 4 processors and 30g of ram as the more threads you give it, the faster it will run.  Also, if the database container is not set to 
```

Comment:
If working in a cloud based environment, such as the AWS CloudFormations on the Senzing GitHub, you will want to max out any containers used to run these utilities to at least 4 cores and 30GB of RAM. An example would be the SSHD container task in AWS created by the Senzing CloudFormations. This provides additional resources and threads providing higher throughput.

Reply:
not done, there is so much more to say here that should be documented elsewhere.   They need to max out the container and database